### PR TITLE
Document how to override templates

### PR DESCRIPTION
**Pull request by psrpinto**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,17 +65,12 @@ function oidc_clients() {
...
+## Override templates
+You can override templates provided by this plugin from a theme. To do so, you should create an `openid-connect/` directory under your theme's root:
 
 ```shell
-wp-env start
+wp-content/themes/my-theme/
+ │── openid-connect/
+ │──── main.php
+ └──── forbidden.php
 ```
```
Comment (ashfame) [inline README.md:76]: I think it would be great to also link the Github for that class we are using here, so that folks can learn more about it & a heads up for those familiar with it already.

**Review by ashfame — APPROVED** [review]
Looks good!

File: README.md (hunk 1 above)
Comment (psrpinto) [reply README.md:76]: I think linking to the docs of [Gamajo-Template-Loader](https://github.com/GaryJones/Gamajo-Template-Loader) could be confusing to our users, who could be non-technical or "half-technical", as those docs are more targeted at plugin developers, and contain a lot of information that is not that relevant when you just want to override a template.

Additionally, the fact that we're using Gamajo-Template-Loader is a detail of the current implementation, in the future we could move away from using it. Linking to their docs might give the impression to our users that Gamajo-Template-Loader is part of our public API.

I think it's best that consumers of this plugin only interact with the templating system through `Templating.php`, as that gives us the possibility of changing the internals while maintaining compatibility, as long as we don't change its public API.

I improved the docs on overriding templates so that it's more clear how to do it. Does this make sense @ashfame? Do you think the current docs are clear and enough for someone who wants to override a template?

Comment (ashfame) [reply README.md:78]: Ok, that makes sense! Yes, the instructions are quite detailed :+1: 

I was only thinking about mentioning support for child/parent theme when I made that comment, as in, order of overriding templates `childtheme > parenttheme > plugin`.

**Review by ashfame — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (psrpinto) [reply README.md:78]: Makes sense. I pushed a commit to mention parent and child themes.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+Start local site with:
+
+```shell
+wp-env start
+```
+
+Change permalink structure:
+
+```shell
+wp-env run cli "wp rewrite structure '/%postname%'"
```
Comment (akirk) [inline CONTRIBUTING.md:13]: This is a pretty big thing to ask for. Why is this necessary?

File: README.md [hunk 3]
```diff
@@ -65,17 +65,47 @@ function oidc_clients() {
 }
 ```
 
-## Development
-Start local site with:
+## Overriding templates
+The pages provided by this plugin are rendered using templates. If you so wish, you can use your own templates instead of this plugins's [default templates](templates).
+
+To do so, you should create an `openid-connect/` directory under your theme, containing only the templates you wish to override. For example, if you wanted to override `authenticate/main.php` and `authenticate/forbidden.php` you would create them under an `openid-connect/` directory in your theme:
```
Comment (akirk) [inline README.md:71]: I am confused by the use of the directory `authenticate/`. It is not necessary when overriding the templates? Why do we have this subdirectory in the first place?

File: CONTRIBUTING.md (hunk 2 above)
Comment (psrpinto) [reply CONTRIBUTING.md:13]: This is something I noticed is needed at the moment, so I documented it. There's an issue for it at https://github.com/Automattic/wp-openid-connect-server/issues/19, once that is fixed these instructions can be removed.

File: README.md (hunk 3 above)
Comment (psrpinto) [reply README.md:71]: `authenticate` is the page for which the templates are, i.e. the page with URI `/openid-connect/authenticate/`. So there is one subdirectory per page.

At the moment there is only one page, but there might be other pages in the future, for example a page where you remove previously granted consent. Each page would get its own subdirectory under `templates/`.

Starting with subdirectories since day one, even if there is only one page at the moment, allows us to not break backwards compatibility once (or if) we do introduce other pages. Because users can override these templates, moving them would break backwards compatibility.

Comment (psrpinto) [reply README.md:71]: Some of the paths in the docs were wrong here, I fixed that in https://github.com/Automattic/wp-openid-connect-server/pull/33
